kotlin/display-layer-view-state/src/main/java/com/esri/arcgisruntime/sample/displaylayerviewstate/MainActivity.kt

```diff
@@ -0,0 +1,151 @@
+      setViewpoint(Viewpoint(Point(-11e6, 45e5, SpatialReferences.getWebMercator()), 40_000_000.0))
+    }
+
+    mapView.addLayerViewStateChangedListener { layerViewStateChangedEvent ->
+      // get the layer which changed its state
+      val layer = layerViewStateChangedEvent.layer
+      // we only want to check the view state of the image layer
+      if (layer != featureLayer) {
+        return@addLayerViewStateChangedListener
+      }
+
+      val layerViewStatus = layerViewStateChangedEvent.layerViewStatus
+      // if there is an error or warning, display it in a toast
+      layerViewStateChangedEvent.error?.let { error ->
+        val message = error.cause?.toString() ?: error.message
+        Toast.makeText(this, message, Toast.LENGTH_LONG).show()
+      }
+      displayViewStateText(layerViewStatus)
+    }
+
+    loadButton.setOnClickListener {
+      if (featureLayer != null) {
+        return@setOnClickListener
+      }
+      // load a feature layer from a portal item
+      val portalItem = PortalItem(
+        Portal("https://runtime.maps.arcgis.com/"),
+        "b8f4033069f141729ffb298b7418b653"
+      )
+      featureLayer = FeatureLayer(portalItem, 0).apply {
+        // set the scales at which this layer can be viewed
+        minScale = 400_000_000.0
+        maxScale = minScale / 10
+      }
+      // add the layer on the map to load it
+      mapView.map.operationalLayers.add(featureLayer)
+      // hide the button
+      loadButton.apply {
+        isEnabled = false
+        visibility = View.GONE
+      }
+      // show the view state UI and the hide layer button
+      statesContainer.visibility = View.VISIBLE
+      hideButton.visibility = View.VISIBLE
+    }
+
+    hideButton.setOnClickListener {
+      featureLayer?.apply {
+        // toggle visibility
+        isVisible = !isVisible
+        // update button text
+        hideButton.text = when {
+          isVisible -> getString(R.string.show_layer)
+          else -> getString(R.string.hide_layer)
+        }
+      }
+    }
+  }
+
+  /**
+   * Formats and displays the layer view status flags in a text view.
+   *
+   * @param layerViewStatus to display
+   */
+  private fun displayViewStateText(layerViewStatus: EnumSet<LayerViewStatus>) {
+    // for each view state property that's active,
+    // add it to a list and display the states as a comma-separated string
+    val stringList = mutableListOf<String>()
+    if (layerViewStatus.contains(LayerViewStatus.ACTIVE)) {
+      stringList.add(getString(R.string.activeStateTextViewString))
+    }
+    if (layerViewStatus.contains(LayerViewStatus.ERROR)) {
+      stringList.add(getString(R.string.errorStateTextViewString))
+    }
+    if (layerViewStatus.contains(LayerViewStatus.LOADING)) {
+      stringList.add(getString(R.string.loadingStateTextViewString))
+    }
+    if (layerViewStatus.contains(LayerViewStatus.NOT_VISIBLE)) {
+      stringList.add(getString(R.string.notVisibleStateTextViewString))
```

Comment:
these names can be simplified and moved to snake case too.

not_visible_state or something like that